wagtail/admin/views/account.py

```diff
@@ -44,80 +49,207 @@ def password_reset_enabled():
+    name = 'notifications'
+    title = gettext_lazy('Notifications')
+    tab = notifications_tab
+    order = 100
+    form_class = NotificationPreferencesForm
+    form_object = 'profile'
+
+    def is_active(self):
+        # Hide the panel if the user can't edit or publish pages
+        user_perms = UserPagePermissionsProxy(self.request.user)
+        if not user_perms.can_edit_pages() and not user_perms.can_publish_pages():
+            return False
+
+        # Hide the panel if there are no notification preferences
+        return self.get_form().fields
+
+
+class LocaleSettingsPanel(BaseSettingsPanel):
+    name = 'locale'
+    title = gettext_lazy('Locale')
+    order = 400
+    form_class = LocalePreferencesForm
+    form_object = 'profile'
+
+    def is_active(self):
+        return len(get_available_admin_languages()) > 1 or len(get_available_admin_time_zones()) > 1
+
+
+class ChangePasswordPanel(BaseSettingsPanel):
+    name = 'password'
+    title = gettext_lazy('Password')
+    order = 500
+    form_class = PasswordChangeForm
+
+    def is_active(self):
+        return password_management_enabled() and self.user.has_usable_password()
+
+    def get_form(self):
+        # Note: don't bind the form unless a field is specified
+        # This prevents the validation error from displaying if the user wishes to ignore this
+        bind_form = False
+        if self.request.method == 'POST':
+            bind_form = any([
+                self.request.POST.get(self.name + '-old_password'),
+                self.request.POST.get(self.name + '-new_password1'),
+                self.request.POST.get(self.name + '-new_password2'),
+            ])
+
+        if bind_form:
+            return self.form_class(self.user, self.request.POST, prefix=self.name)
+        else:
+            return self.form_class(self.user, prefix=self.name)
 
-    return TemplateResponse(request, 'wagtailadmin/account/change_password.html', {
-        'form': form,
-        'can_change_password': can_change_password,
-    })
 
+# Views
 
-def change_email(request):
-    if not email_management_enabled():
-        raise Http404
-    if request.method == 'POST':
-        form = EmailForm(request.POST, instance=request.user)
+@sensitive_post_parameters()
+def account(request):
+    # Fetch the user and profile objects once and pass into each panel
+    # We need to use the same instances for all forms so they don't overwrite each other
+    user = request.user
+    profile = UserProfile.get_for_user(request.user)
+
+    # Panels
+    panels = [
+        NameEmailSettingsPanel(request, user, profile),
+        AvatarSettingsPanel(request, user, profile),
+        NotificationsSettingsPanel(request, user, profile),
+        LocaleSettingsPanel(request, user, profile),
+        ChangePasswordPanel(request, user, profile),
+    ]
+    for fn in hooks.get_hooks('register_account_settings_panel'):
+        panel = fn(request, user, profile)
+        if panel:
```

Comment:
```suggestion
        if panel and panel.is_active():
```

which would reduce the need for `panels = [panel for panel in panels if panel.is_active()]` below